runtime/gateway.go

```diff
@@ -125,22 +127,26 @@ func CreateGateway(
 ) (*Gateway, error) {
 	var metricsBackend tally.CachedStatsReporter
 	var logWriter zapcore.WriteSyncer
+	var scopeExtractors []ContextScopeExtractor
 	if opts != nil && opts.MetricsBackend != nil {
 		metricsBackend = opts.MetricsBackend
 	}
 	if opts != nil && opts.LogWriter != nil {
 		logWriter = opts.LogWriter
 	}
+	if opts != nil && opts.ScopeExtractors != nil {
```

Comment:
This code is a bit redundant. Can you change it to be something like 

```
if opts != nil {
     opts = &Options{}
}
```

Since the other nil checks and local variable assignments are redundant; if this nil check doesn't pass then `scopeExtractors` is still nil...